src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -2600,5 +2623,273 @@ private void Remove(int startIndex, int count, out StringBuilder chunk, out int
+                        {
+                            if ((uint)charsWritten > (uint)destination.Length)
+                            {
+                                // Protect against faulty ISpanFormattable implementations returning invalid charsWritten values.
+                                FormatError();
+                            }
+
+                            _stringBuilder.m_ChunkLength += charsWritten;
+                        }
+                        else
+                        {
+                            // Not enough room in the current chunk.  Take the slow path that formats into temporary space
+                            // and then copies the result into the StringBuilder.
+                            AppendFormattedWithTempSpace(value, 0, format);
+                        }
+                    }
+                    else
+                    {
+                        _stringBuilder.Append(((IFormattable)value).ToString(format, _provider)); // constrained call avoiding boxing for value types
+                    }
+                }
+                else if (value is not null)
+                {
+                    _stringBuilder.Append(value.ToString());
+                }
+            }
+
+            /// <summary>Writes the specified value to the builder.</summary>
+            /// <param name="value">The value to write.</param>
+            /// <param name="alignment">Minimum number of characters that should be written for this value.  If the value is negative, it indicates left-aligned and the required minimum is the absolute value.</param>
+            public void AppendFormatted<T>(T value, int alignment) =>
+                // If this overload is being used, it's almost certainly a non-0 alignment, so just fall back to
+                // formatting into temporary space and then appending that result along with any required padding.
+                AppendFormattedWithTempSpace(value, alignment, format: null);
+
+            /// <summary>Writes the specified value to the builder.</summary>
+            /// <param name="value">The value to write.</param>
+            /// <param name="format">The format string.</param>
+            /// <param name="alignment">Minimum number of characters that should be written for this value.  If the value is negative, it indicates left-aligned and the required minimum is the absolute value.</param>
+            public void AppendFormatted<T>(T value, int alignment, string? format)
+            {
+                if (alignment == 0)
+                {
+                    // This overload is used as a fallback from several disambiguation overloads, so special-case 0
+                    // to avoid using temporary space if we don't have to.
+                    AppendFormatted(value, format);
+                }
+                else
```

Comment:
Would 

```C#
else if (alignment < 0)
{
    int start = _builder.Length;
    AppendFormatted(value, format);
    int end = _builder.Length;
    int pad = alignment - end - start;
    // Whatever to do to Append `pad` spaces here
}
```

be more space-and-time efficient?  The left pad shouldn't need to rent, and either way it avoids the double copy.

I'm perfectly willing to believe "no, it's not more efficient", just asking.

Reply:
Yeah, it's reasonable.